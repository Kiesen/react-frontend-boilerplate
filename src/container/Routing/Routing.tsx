import React, { FC } from 'react';
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';

import Login from 'container/Login/Login';
import Home from 'container/Home/Home';
import AuthRoute from 'components/hoc/AuthRoute';
import { BASE, LOGIN } from 'constants/routes';

/**
 * The routing container component renders all possible
 * frontend routes wrapped by BrowserRouter and Switch component.
 *
 * @returns Routes wrapped by BrowserRouter and Switch
 */
const Routing: FC<{}> = () => (
  <BrowserRouter>
    <Switch>
      <Route exact path={LOGIN} component={Login} />

      <AuthRoute path={BASE} component={Home} />

      <Redirect to={BASE} />
    </Switch>
  </BrowserRouter>
);

export default Routing;
