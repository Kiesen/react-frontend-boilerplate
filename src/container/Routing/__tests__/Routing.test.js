import React from 'react';
import { shallow } from 'enzyme';

import Routing from 'container/Routing/Routing';

describe('<Routing />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Routing />);
    expect(wrapper).toMatchSnapshot();
  });
});
