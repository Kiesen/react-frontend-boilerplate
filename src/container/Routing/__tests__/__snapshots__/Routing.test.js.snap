// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routing /> renders correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <AuthRoute
      component={[Function]}
      path="/"
    />
    <Redirect
      to="/"
    />
  </Switch>
</BrowserRouter>
`;
